### ISSUE TITLE
[building-rollup] injectServiceWorker: Get service worker file name from Workbox config

### ISSUE THREAD

**Issue by abdonrd** [open-wc/open-wc#1564]
Right now the service worker file name from the `injectServiceWorker` option is hardcoded here:

https://github.com/open-wc/open-wc/blob/417d890c7c2b7574168191df05c76d428fb01ca8/packages/building-rollup/src/utils.js#L53

It would be nice if we can get the file name from the Workbox `swDest` config.
Otherwise, it doesn't work when the user customizes their Workbox settings.

**Comment by thepassle** [comment]
I agree the sw name should be customizable, but I'm not sure if we should get this from the workbox config. It feels like 'magic' behavior to me, and is prone to lead to confusion. I think it's better to be explicit.

**Comment by abdonrd** [comment]
The bad part of that, is that you would have to define the same file name in two places.

**Comment by abdonrd** [comment]
Although it could be that you create the SW without using Workbox!

**Comment by LarsDenBakker** [comment]
I actually think it makes sense to use the `swDest` option for this. Ideally `rollup-plugin-workbox` would just take a service worker file name and emit this as a rollup asset.

We can merge the user config with the default config for workbox, and take the `swDest` path from there as input for the injection logic stripping off the output directory. This way there is a single source of truth, and it's easy to customize by a user through the existing means:

```js
createSpaConfig({
  workbox: {
    swDest: path.join(process.cwd(), 'dist', 'my-service-worker.js'),
  }
})
```

What do you think @thepassle ?

**Comment by thepassle** [comment]
In that case I think its fine, because the user has already opted in to using their own workbox configuration so it makes sense that the service worker registration uses the `swDest` from it, but I dont think we should try to find a workbox.config.js in the users project, and _then_ magically use that

**Comment by LarsDenBakker** [comment]
Yep, agree with that!

**Comment by abdonrd** [comment]
Oh, yes! My config is like this, not from a `workbox.config.js`:

```javascript
const DIST_PATH = 'server/dist/';

const workboxConfig = {
  // ...
  swDest: `${DIST_PATH}service-worker.js`
};

const baseConfig = createSpaConfig({
  outputDir: DIST_PATH,
  legacyBuild: true,
  workbox: workboxConfig,
  injectServiceWorker: true
});
```

**Comment by abdonrd** [comment]
Related to this code, I think this Terser.minify is not needed:

https://github.com/open-wc/open-wc/blob/417d890c7c2b7574168191df05c76d428fb01ca8/packages/building-rollup/src/utils.js#L49-L62

This works in the same way:

```javascript
/**
 * @param {string} htmlString
 * @returns {string}
 */
function applyServiceWorkerRegistration(htmlString) {
  const documentAst = parse(htmlString);
  const body = query(documentAst, predicates.hasTagName('body'));
  const scriptCode = `
    if ('serviceWorker' in navigator) {
      window.addEventListener('load', function() {
        navigator.serviceWorker
          .register('./sw.js')
          .then(function() {
            console.log('ServiceWorker registered.');
          })
          .catch(function(err) {
            console.log('ServiceWorker registration failed: ', err);
          });
      });
    }
  `;
  const swRegistration = createScript({}, scriptCode);

  append(body, swRegistration);
  return serialize(documentAst);
}
```
